src/test/shared/clients/devenvClient.test.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { SinonStub, SinonStubbedInstance, SinonStubbedMember, createSandbox, createStubInstance, SinonSpiedMember } from 'sinon'
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+
+import {ExtensionUserActivity} from '../../../shared/extensionUtilities'
+import { DevEnvClient, DevEnvActivity } from '../../../shared/clients/devenvClient'
+import { sleep } from '../../../shared/utilities/timeoutUtils'
+
+
+describe('DevEnvActivity', function () {
+    let triggerUserActivityEvent: (obj: any) => Promise<void>
+    let userActivityEvent: SinonStub<Parameters<vscode.Event<any>>, vscode.Disposable>
+    let activitySubscriber: SinonStubbedMember<(timestamp: number) => void>
+    let devEnvClientStub: SinonStubbedInstance<DevEnvClient>
+    let devEnvActivity: DevEnvActivity
+    let sendActivityUpdateSpy: SinonSpiedMember<DevEnvActivity['sendActivityUpdate']>
+    let sandbox: sinon.SinonSandbox
+
+    before(function() {
+        sandbox = createSandbox()
+    })
+
+    beforeEach(async function () {
+        userActivityEvent = sandbox.stub(
+            vscode.window, 
+            'onDidChangeActiveColorTheme'
+        )
+        userActivityEvent.callsFake((throttledEventFire: (obj: any) => void) => {
+            triggerUserActivityEvent = async (obj: any) => {
+                throttledEventFire(obj)
+                // There are multiple layers of emitters firing
+                // so we wait a bit before returning. Otherwise,
+                // racecondition when waiting for activitySubscriber
+                // to be called
+                await sleep(100)
+            }
+            return {
+                dispose: sandbox.stub()
+            }
+        })
+        
+        devEnvClientStub = createStubInstance(DevEnvClient)
+
+        devEnvActivity = (await DevEnvActivity.instanceIfActivityTrackingEnabled(devEnvClientStub as unknown as DevEnvClient, new ExtensionUserActivity(0, [userActivityEvent])))!
+        sendActivityUpdateSpy = sandbox.spy(devEnvActivity, 'sendActivityUpdate')
```

Comment:
We fully control this. Is a spy really needed? We are using mocks and spies a lot, this should not be the default way of testing things.

Reply:
Good find. This wasn't needed anymore, I've removed the test that uses it.